Sources/GRPCHTTP2Core/Client/GRPCClientStreamHandler.swift

```diff
@@ -148,6 +158,28 @@ extension GRPCClientStreamHandler {
   func handlerRemoved(context: ChannelHandlerContext) {
     self.stateMachine.tearDown()
   }
+
+  func channelInactive(context: ChannelHandlerContext) {
+    switch self.stateMachine.unexpectedInboundClose(reason: .channelInactive) {
+    case .forwardStatus(let status):
+      context.fireChannelRead(self.wrapInboundOut(.status(status, [:])))
+    case .doNothing:
+      context.fireChannelInactive()
+    case .fireError:
+      assertionFailure("`fireError` should only happen on the server side, never on the client.")
+    }
+  }
+
+  func errorCaught(context: ChannelHandlerContext, error: any Error) {
+    switch self.stateMachine.unexpectedInboundClose(reason: .errorThrown(error)) {
+    case .forwardStatus(let status):
+      context.fireChannelRead(self.wrapInboundOut(.status(status, [:])))
+    case .doNothing:
+      context.fireErrorCaught(error)
```

Comment:
Hmm, I'm not sure this is right because we want to avoid the extra errors. Presumably this we `doNothing` when the inbound stream is already finished?